pkgs/os-specific/linux/evdi/evdi_fix_flags.diff

```diff
@@ -0,0 +1,13 @@
+diff --git a/Makefile b/Makefile
+index f343f78..e7526e9 100644
+--- a/Makefile
++++ b/Makefile
+@@ -2,7 +2,7 @@
+ # Copyright (c) 2015 - 2020 DisplayLink (UK) Ltd.
+ #
+ 
+-FLAGS=-Werror -Wextra -Wall -Wmissing-prototypes -Wstrict-prototypes -Wno-error=missing-field-initializers -Werror=sign-compare
++FLAGS=-Wextra -Wall -Wmissing-prototypes -Wstrict-prototypes -Wno-error=missing-field-initializers 
```

Comment:
Try to specify `-Wno-error` in `NIX_CFLAGS_COMPILE` instead. It's less invasive than a full patch.

Reply:
Does not work for me (partial build errors):

```
cc1: some warnings being treated as errors
make[3]: *** [/nix/store/ypfs52m4sfzgzpggy9v85fmqz5y7718n-linux-5.15.13-dev/lib/modules/5.15.13/source/scripts/Makefile.build:277: /build/source/module/evdi_platform_drv.o] Error 1
make[2]: *** [/nix/store/ypfs52m4sfzgzpggy9v85fmqz5y7718n-linux-5.15.13-dev/lib/modules/5.15.13/source/Makefile:1868: /build/source/module] Error 2
make[2]: Leaving directory '/nix/store/ypfs52m4sfzgzpggy9v85fmqz5y7718n-linux-5.15.13-dev/lib/modules/5.15.13/build'
make[1]: *** [Makefile:81: module] Error 2
make[1]: Leaving directory '/build/source/module'
make: *** [Makefile:8: all] Error 2
error: builder for '/nix/store/hvx8vxwlavi9082amav084my987yviyl-evdi-1.10.0.drv' failed with exit code 2;
       last 10 log lines:
       > /nix/store/ypfs52m4sfzgzpggy9v85fmqz5y7718n-linux-5.15.13-dev/lib/modules/5.15.13/source/include/linux/usb.h:2039:56: warning: unused parameter 'ev' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-parameter-Wunused-parameter8;;]
       >  2039 | static inline void usb_led_activity(enum usb_led_event ev) {}
       >       |                                     ~~~~~~~~~~~~~~~~~~~^~
       > cc1: some warnings being treated as errors
       > make[3]: *** [/nix/store/ypfs52m4sfzgzpggy9v85fmqz5y7718n-linux-5.15.13-dev/lib/modules/5.15.13/source/scripts/Makefile.build:277: /build/source/module/evdi_platform_drv.o] Error 1
       > make[2]: *** [/nix/store/ypfs52m4sfzgzpggy9v85fmqz5y7718n-linux-5.15.13-dev/lib/modules/5.15.13/source/Makefile:1868: /build/source/module] Error 2
       > make[2]: Leaving directory '/nix/store/ypfs52m4sfzgzpggy9v85fmqz5y7718n-linux-5.15.13-dev/lib/modules/5.15.13/build'
       > make[1]: *** [Makefile:81: module] Error 2
       > make[1]: Leaving directory '/build/source/module'
       > make: *** [Makefile:8: all] Error 2
       For full logs, run 'nix log /nix/store/hvx8vxwlavi9082amav084my987yviyl-evdi-1.10.0.drv'.
error: 1 dependencies of derivation '/nix/store/bm83x4kii5j765d316yj9vw0hmr08q6m-displaylink-5.5.0-beta-59.118.drv' failed to build
error: 1 dependencies of derivation '/nix/store/iwflqlpl0zhij4f3r57lgbbfxnqd5ma9-kernel-modules.drv' failed to build
error: 1 dependencies of derivation '/nix/store/bnhvkl8m5dp717mq46x7a892g4r1y87p-linux-5.15.13-modules.drv' failed to build
error: 1 dependencies of derivation '/nix/store/9g4hbwx666kz0siav5bws22r71f86hnm-nixos-system-bsMF42-22.05pre-git.drv' failed to build
error: 1 dependencies of derivation '/nix/store/r1vm1d4y8iiqf4vbbg3nsif7ghpbchnl-nixops-machines.drv' failed to build
```

Maybe you have another idea?

Reply:
Ah, try also adding `-Wno-error=sign-compare` -- missed it at first.

Reply:
You are right, I thought it was left out intended as I'm not really familiar with C :D
Fixed and working 👍🏻  thanks, much more elegant solution!